src/utils.cpp

```diff
@@ -83,4 +83,18 @@ WindowResultQueue::GetAvgResult() const {
     return result / statstic_num;
 }
 
+std::string
+format_map(const std::string& str, const std::unordered_map<std::string, std::string>& mappings) {
+    std::string result = str;
+
+    for (const auto& [key, value] : mappings) {
+        size_t pos = result.find("{" + key + "}");
+        while (pos != std::string::npos) {
+            result.replace(pos, key.length() + 2, value);
```

Comment:
```C++
result = result.replace(pos, key.length() + 2, value)
```

Reply:
https://en.cppreference.com/w/cpp/string/basic_string/replace maybe no need care return value

Reply:
OK，I known C++ string is mutable :)